### ISSUE TITLE
API error responses aren't surfaced to SDK error handlers

### ISSUE THREAD

**Issue by zdoc01** [contentstack/contentstack-javascript#35]
Hello!

I've noticed several use cases where API failures do not return the error response (containing the `error_code`, `error_message`, etc.) to the error callback handlers I define. Below are a couple of examples (all tested with `v3.10.1` - latest - of the SDK).

## Content type doesn't exist
```js
const Stack = window.Contentstack.Stack(
      'api_key',
      'delivery_token,
      'environment'
    );

    Stack.ContentType('missing_content_type')
      .Query()
      .findOne()
      .then(console.log, console.error);
```

### API response expected in `console.error`
```json
{"error_message":"The Content Type 'missing_content_type' was not found. Please try again.","error_code":118,"errors":{"uid":["is not valid."]}}
```

### Actual response
Empty string.

## Entry doesn't exist
```js
const Stack = window.Contentstack.Stack(
      'api_key',
      'delivery_token,
      'environment'
    );

    Stack.ContentType('valid_content_type_uid')
      .Entry('missing_entry')
      .fetch()
      .then(console.log, console.error);
```

### API response expected in console.error`
```json
{"error_message":"The requested object doesn't exist.","error_code":141,"errors":{"uid":["is not valid."]}}
```

### Actual response
Empty string.

I believe this may stem from the way in which the underlying `fetch` logic is handling failures. The custom `Request` wrapper [rejects these failures with the `response.statusText`](https://github.com/contentstack/contentstack-javascript/blob/master/src/core/lib/request.js#L57) instead of the response JSON.

**Comment by uttamukkoji** [comment]
Thanks, @zdoc01, for raising the issue. We’re looking into it and we’ll push a fix as early as possible.

**Comment by uttamukkoji** [comment]
We have resolved the issue, and will be available in the upcoming version.
Here is pull request #36

**Comment by zdoc01** [comment]
Awesome! Thank you so much @uttamukkoji . Looking forward to the next release =D

**Comment by uttamukkoji** [comment]
@zdoc01, We’ve released a new version v3.11.0 with this issue resolved. We will now proceed to close this thread. If there are further questions regarding this matter, please tag me in your reply. We will gladly continue the discussion and we will reopen the issue
